faiss/IndexIDMap.cpp

```diff
@@ -102,6 +102,24 @@ struct ScopedSelChange {
     }
 };
 
+// RAII object to reset the id_map parameter in ResultCollectorFactory object
+// This object make sure to reset the id_map parameter in ResultCollectorFactory
+// once the program exist current method scope.
+struct ScopedColChange {
```

Comment:
Does this have any threading implications?

Reply:
There is no.